[addon] Switch to performance profile

fetzerch opened the pull request:
Make this addon use the performance profile of bsnes-mercury.
I've modified the CMakeLists.txt for bsnes-mercury to optionally accept a BSNES_PROFILE parameter. That way we can use the same file in all 3 versions of the addon and only put a different flags.txt file.

@garbear: Can you review this change first, then ll push the balanced version afterwards.

fetzerch committed:
[addon] Switch to performance profile

garbear commented on game.libretro.bsnes-mercury-performance/addon.xml:
```diff
@@ -1,14 +1,14 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<addon id="game.libretro.bsnes-mercury" name="SNES (bSNES Mercury)" version="0.94.0" provider-name="Libretro">
+<addon id="game.libretro.bsnes-mercury-performance" name="SNES (bSNES Mercury Performance)" version="0.94.0" provider-name="Libretro">
```
How about the name `SNES (bSNES Mercury "Performance")` to show that "Performance" is the name of a profile instead of the emulator?

garbear commented on game.libretro.bsnes-mercury-performance/addon.xml:
```diff
@@ -18,6 +18,8 @@
     <summary lang="en">bSNES Mercury v0.94</summary>
     <description>bSNES Mercury is a fork of Higan (originally called bSNES), notable for emulating the original hardware as accurately as possible through low-level, cycle-accurate emulation.
 
+Performance profile focussing more on execution speed than on accuracy.
```
Change to `The "Performance" profile focusses...` please

fetzerch committed:
[cmake] Remove unused DEBLIBS variable

This addon does not build an own addon target, but only compiles the
libretro core. Hence there is no need to specify any DEPLIBS.
Additionally build_addon only uses the first two entries in the
_CUSTOM_BINARY list (from, to), so that the third entry can be removed.
